mediator/esmFldsExchange_nems_mod.F90

```diff
@@ -73,26 +83,28 @@ subroutine esmFldsExchange_nems(gcomp, phase, rc)
     call addfld(fldListFr(compocn)%flds, 'So_omask')
     call addmap(fldListFr(compocn)%flds, 'So_omask', compice,  mapfcopy, 'unset', 'unset')
 
-    ! atm and ocn fields required for atm/ocn flux calculation'
-    allocate(flds(6))
-    flds = (/'Sa_u   ','Sa_v   ','Sa_z   ','Sa_tbot','Sa_pbot','Sa_shum'/)
-    do n = 1,size(flds)
-       fldname = trim(flds(n))
-       call addfld(fldListFr(compatm)%flds, trim(fldname))
-       call addmap(fldListFr(compatm)%flds, trim(fldname), compocn, mapnstod_consf, 'none', 'unset')
-    end do
-    deallocate(flds)
-
-    ! unused fields needed by the atm/ocn flux computation
-    allocate(flds(13))
-    flds = (/'So_tref  ', 'So_qref  ','So_u10   ', 'So_ustar ','So_ssq   ', &
-             'So_re    ', 'So_duu10n','Faox_lwup', 'Faox_sen ','Faox_lat ', &
-             'Faox_evap', 'Faox_taux','Faox_tauy'/)
-    do n = 1,size(flds)
-       fldname = trim(flds(n))
-       call addfld(fldListMed_aoflux%flds, trim(fldname))
-    end do
-    deallocate(flds)
+    if ( trim(coupling_mode) == 'nems_orig_data') then
```

Comment:
Can this be in xml?

Reply:
Not yet. We need a longer term plan for now to migrate the esmFldsExchange_xxx.F90 to either xml or yaml and ingest the data. We also have the challenge that we cannot be assured of getting the coupling_mode from the drivers - since we have not put in any real requirements yet for the drivers.